### PR TITLE
Add ActionMailer adapter

### PR DESCRIPTION
**Pull request by alexshgov**
* Add `ActionMailer` adapter with appropriate specs
* Update readme

### CODE REVIEW

**Comment by palkan** [comment]
Please, rebase on top of master.
No need to add `actionmailer` to gemfiles, now we add the whole Rails.

File: lib/isolator/errors.rb [hunk 1]
```diff
@@ -18,4 +18,8 @@ class BackgroundJobError < UnsafeOperationError
     MESSAGE = "You are trying to enqueue background job inside db transaction." \
       "In case of transaction failure, this may lead to data inconsistency and unexpected bugs"
   end
+
+  class ActionMailerError < UnsafeOperationError
+    MESSAGE = "You are trying to send email inside db transaction. "
```
Comment (palkan) [inline lib/isolator/errors.rb]: Trailing space

File: spec/isolator/orm_adapters/active_record_spec.rb [hunk 2]
```diff
@@ -87,6 +87,40 @@
         end
       end
     end
+
+    context "Email sending" do
+      context "ActionMailer" do
```
Comment (palkan) [inline spec/isolator/orm_adapters/active_record_spec.rb]: Let's add the following case:
-`:active_job` adapter is disabled
- `deliver_later` is called
- **no error** is raised with a transaction.

I want to make sure, that `action_mailer` adapter detects only delivery itself and not background job enqueueing.

Another case – email rendering without sending, i.e. `SampleEmail.hello.body` should not raise errors.

Comment (alexshgov) [reply spec/isolator/orm_adapters/active_record_spec.rb]: @palkan done

File: lib/isolator/adapters/action_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+ActiveSupport.on_load(:action_mailer) do
+  Isolator.isolate :action_mailer, ActionMailer::Base.singleton_class,
```
Comment (palkan) [inline lib/isolator/adapters/action_mailer.rb]: What if make this adapter _Rails-less_ and directly track `mail` gem (`Mail::Message#deliver` method)?

That should work with other frameworks too (like Hanami), 'cause everyone uses `mail` under the hood.

No need to rewrite specs, btw; maybe, add a _raw_ `Mail::Message` example.

File: lib/isolator/errors.rb [hunk 4]
```diff
@@ -18,4 +18,8 @@ class BackgroundJobError < UnsafeOperationError
     MESSAGE = "You are trying to enqueue background job inside db transaction. " \
       "In case of transaction failure, this may lead to data inconsistency and unexpected bugs"
   end
+
+  class ActionMailerError < UnsafeOperationError
```
Comment (palkan) [inline lib/isolator/errors.rb]: Let's use more general naming, e.g. `MailerError`

File: lib/isolator/adapters/action_mailer.rb (hunk 3 above)
Comment (alexshgov) [reply lib/isolator/adapters/action_mailer.rb]: @palkan  Good point, I'll try!

Comment (alexshgov) [reply lib/isolator/adapters/action_mailer.rb]: @palkan Check it out https://github.com/palkan/isolator/pull/5/commits/92f9329758a1fcf2bf550a8f6b56f1fe99a5f9e1

**Comment by palkan** [comment]
Thanks!

**Pull request merged**